src/components/organisms/test/hooks/use-test.ts

```diff
@@ -1,82 +1,86 @@
+  const testProcess = useAppSelector(selectTestProcess);
+  const isTestCreated = useAppSelector(selectProcessCreationTest);
+  const testResult = useAppSelector(selectTestResult);
 
-  // TODO Функционал на пересдачу теста
-  // TODO https://github.com/Studio-Yandex-Practicum/lizaalert_frontend/issues/422
-  const retake = () => null;
+  const isLoading = LOADING_PROCESS_MAP[testProcess];
 
-  // TODO: настроить условия для percentArr.push, значений percent и checkedCount в связи с новой логикой валидации ответов с бэка, настроить условия для нового типа ответа 'text_answer'
-  useEffect(() => {
-    if (test.passing_score && test.questions && test.questions.length >= 0) {
-      const percentArr: number[] = [];
-
-      test.questions.forEach((question) => {
-        if (question.question_type === Controls.RADIO) {
-          question.content.forEach(() => {
-            percentArr.push(100);
-          });
-        } else {
-          const weight = 100 / question.content.length;
-          let percent = 0;
-          question.content.forEach(() => {
-            percent += weight;
-          });
-          percentArr.push(percent);
-        }
-      });
-
-      const middlePercent =
-        percentArr.reduce((sum, percent) => sum + percent, 0) /
-        percentArr.length;
+  const dispatch = useAppDispatch();
 
-      const resultPercent = Math.round(middlePercent);
-
-      setTestResultPercent(resultPercent);
-      setIsSuccess(resultPercent >= test.passing_score);
+  // TODO: Функционал на пересдачу теста
+  // TODO: https://github.com/Studio-Yandex-Practicum/lizaalert_frontend/issues/422
+  const retake = () => {
+    if (lessonId) {
+      void dispatch(createTest(lessonId));
     }
-  }, [test.passing_score, test.questions]);
-
-  const handleButtonDisabledState = () => {
-    let isDisabled = false;
+    dispatch(updateAnswerReset());
+    setIsSuccess(false);
+    setTestResultPercent(0);
+    setIsSubmitted(false);
+  };
 
-    if (test.questions?.length) {
-      test.questions.forEach((question) => {
-        let checkedCount = 0;
+  const createNewTest = async () => {
+    if (lessonId) {
+      await dispatch(createTest(lessonId));
+    }
+  };
 
-        question.content.forEach(() => {
-          checkedCount += 1;
-        });
+  const sendTestOnValidation = async (): Promise<void> => {
+    if (lessonId && isTestCreated) {
+      await dispatch(validateTest({ id: lessonId, answers }));
+      setIsSubmitted(true);
+    }
+  };
 
-        if (checkedCount === 0) isDisabled = true;
-      });
+  useEffect(() => {
+    if (testResult) {
+      const percent = calculatePercent(
+        testResult.answers.length,
+        testResult.score
+      );
+      setTestResultPercent(percent);
+      setIsSuccess(percent >= AVERAGE_TEST_RESULT);
     }
+  }, [testResult]);
 
-    return isDisabled;
-  };
+  const handleButtonDisabledState = () =>
+    isTestCreated === 'Failed' ||
```

Comment:
```suggestion
    isTestCreated === ProcessEnum.Failed ||
```

или сразу булеву поставить, если определить ее выше:

```suggestion
    isTestCreated ||
```